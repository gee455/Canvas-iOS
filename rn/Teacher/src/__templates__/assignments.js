//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: 1376,
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment()],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
