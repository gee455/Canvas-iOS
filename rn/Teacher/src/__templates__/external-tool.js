//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const externalTool: Template<AddressBookResult> = template({
  id: '1',
  name: 'LTI FTW',
  url: 'https://canvas.instructure.com/lti',
})

export const ltiLaunchDefinition: Template<LtiLaunchDefinition> = template({
  definition_id: 42,
  placements: {
    course_navigation: {
      url: 'https://rollcall.instructure.com/launchy-launch',
    },
  },
})
