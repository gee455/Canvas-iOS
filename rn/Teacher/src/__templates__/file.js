//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const usageRights: Template<UsageRights> = template({
  legal_copyright: '',
  use_justification: 'own_copyright',
})

export const file: Template<File> = template({
  id: '111',
  display_name: 'Book Report',
  url: 'http://canvaslms.com/bookreport',
  size: 488212,
  thumbnail_url: 'http://fillmurray.com/322/200',
  mime_class: 'document',
  preview_url: '/',
  filename: '/bookreport',
  parent_folder_id: null,
  locked: false,
  hidden: false,
  unlock_at: null,
  lock_at: null,
  usage_rights: usageRights(),
  'content-type': 'application/document',
})
