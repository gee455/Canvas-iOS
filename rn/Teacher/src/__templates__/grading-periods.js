//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const gradingPeriod: Template<GradingPeriod> = template({
  id: '1023',
  title: 'First Block',
  start_date: '2014-01-07T15:04:00Z',
  end_date: '2014-05-07T17:07:00Z',
  close_date: '2014-06-07T17:07:00Z',
  weight: 33.33,
})
