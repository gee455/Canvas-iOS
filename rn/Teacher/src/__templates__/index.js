//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
