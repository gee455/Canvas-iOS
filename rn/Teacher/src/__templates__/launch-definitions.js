//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const launchDefinitionGlobalNavigationItem: Template<ExternalToolLaunchDefinitionGlobalNavigationItem> = template({
  message_type: 'basic-lti-launch-request',
  url: 'https://mobiledev.beta.instructuremedia.com/lti/launch?custom_gauge_launch_type=global_nav',
  title: 'Gauge',
})

export const launchDefinition: Template<ExternalToolLaunchDefinition> = template({
  definition_type: 'ContextExternalTool',
  definition_id: '360860',
  name: 'Gauge',
  description: 'Tools for Education',
  domain: 'gauge.instructure.com',
  placements: {
    global_navigation: launchDefinitionGlobalNavigationItem(),
  },
})
