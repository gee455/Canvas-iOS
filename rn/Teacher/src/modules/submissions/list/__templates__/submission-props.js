//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
