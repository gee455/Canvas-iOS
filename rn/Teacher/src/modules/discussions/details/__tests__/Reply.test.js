//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/* @flow */

import { shallow } from 'enzyme'
import React from 'react'
import { ActionSheetIOS } from 'react-native'
import Reply, { type Props } from '../Reply'
import httpClient from '../../../../canvas-api/httpClient'
import * as template from '../../../../__templates__'

jest.mock('ActionSheetIOS', () => ({
  showActionSheetWithOptions: jest.fn(),
}))

describe('DiscussionReplies', () => {
  let props: Props
  beforeEach(() => {
    let reply = template.discussionReply({ id: '1' })
    reply.replies = [template.discussionReply({ id: 2 })]
    let user = template.userDisplay()
    props = {
      reply: reply,
      depth: 0,
      readState: 'read',
      participants: { [user.id]: user },
      context: 'courses',
      contextID: '1',
      discussionID: '1',
      entryID: '1',
      deleteDiscussionEntry: jest.fn(),
      replyToEntry: jest.fn(),
      myPath: [0],
      navigator: template.navigator(),
      onPressMoreReplies: jest.fn(),
      maxReplyNodeDepth: 2,
      discussionLockedForUser: false,
      rating: null,
      showRating: false,
      canRate: false,
      rateEntry: jest.fn(),
      isLastReply: false,
    }
    jest.clearAllMocks()
  })

  it('renders', () => {
    expect(shallow(<Reply {...props}/>)).toMatchSnapshot()
  })

  it('renders deleted', () => {
    props.reply.deleted = true
    let tree = shallow(<Reply {...props} />)
    let webview = tree.find('RichContent')
    expect(webview.props().html.includes('Deleted this reply.')).toEqual(true)
  })

  it('renders with no user', () => {
    props.reply.user_id = ''
    props.reply.editor_id = ''
    let tree = shallow(<Reply {...props} />)
    expect(tree.find('Avatar').props().userName).toEqual('?')
  })

  it('renders with closed discussion as student', () => {
    props.discussionLockedForUser = true
    let tree = shallow(<Reply {...props} />)

    expect(tree.find('[testID="discussions.reply-btn"]').length).toEqual(0)
    expect(tree.find('[testID="discussions.edit-btn"]').length).toEqual(0)
    expect(tree.find('[testID="discussion.reply.rate-btn"]').length).toEqual(0)
  })

  it('renders likes even if closed as student', () => {
    props.discussionLockedForUser = true
    props.showRating = true
    props.canRate = true
    let tree = shallow(<Reply {...props} />)

    expect(tree.find('[testID="discussion.reply.rate-btn"]').length).toEqual(1)
  })

  it('navigates to the context card when an avatar is pressed', () => {
    let tree = shallow(<Reply {...props} />)
    let avatar = tree.find('[testID="reply.avatar"]')
    avatar.simulate('press')
    expect(props.navigator.show).toHaveBeenCalledWith(
      `/courses/1/users/1`,
      { modal: true },
    )
  })

  it('actionMore', () => {
    props.depth = 2
    props.reply.replies = [template.discussionReply({ id: 2 }), template.discussionReply({ id: 3 }), template.discussionReply({ id: 4 })]
    let tree = shallow(<Reply {...props} />)
    let replyButton = tree.find('[testID="discussion.more-replies"]')
    replyButton.simulate('press')
    expect(props.onPressMoreReplies).toHaveBeenCalledWith([0])
  })

  it('renders more button with some deleted replies', () => {
    let a = template.discussionReply({ id: '2' })
    let b = template.discussionReply({ id: '3' })
    let c = template.discussionReply({ id: '4', deleted: true })

    let reply = template.discussionReply({ id: '1', replies: [a, b, c] })
    props.reply = reply
    props.depth = 2
    let tree = shallow(<Reply {...props} />)
    expect(tree.find('[testID="discussion.more-replies"]').length).toEqual(1)
  })

  it('edit action sheet calls delete', () => {
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((_, cb) => cb(1))
    let tree = shallow(<Reply {...props} />)
    let edit = tree.find('[testID="discussion.edit-btn"]')
    edit.simulate('press')

    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalled()
    expect(props.deleteDiscussionEntry).toHaveBeenCalledWith('courses', '1', '1', '1', [0])
  })

  it('edit action sheet calls cancel', () => {
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((_, cb) => cb(2))
    let tree = shallow(<Reply {...props} />)
    let edit = tree.find('[testID="discussion.edit-btn"]')
    edit.simulate('press')

    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalled()
    expect(props.deleteDiscussionEntry).not.toHaveBeenCalled()
    expect(props.navigator.show).not.toHaveBeenCalled()
  })

  it('edit action sheet calls edit', () => {
    ActionSheetIOS.showActionSheetWithOptions = jest.fn((_, cb) => cb(0))
    let tree = shallow(<Reply {...props} />)
    let edit = tree.find('[testID="discussion.edit-btn"]')
    edit.simulate('press')

    expect(ActionSheetIOS.showActionSheetWithOptions).toHaveBeenCalled()
    expect(props.navigator.show).toHaveBeenCalledWith('/courses/1/discussion_topics/1/reply', { modal: true }, { message: props.reply.message, entryID: props.reply.id, isEdit: true, indexPath: props.myPath })
  })

  it('reply to entry', () => {
    let tree = shallow(<Reply {...props} />)
    let reply = tree.find('[testID="discussion.reply-btn"]')
    reply.simulate('press')

    expect(props.replyToEntry).toHaveBeenCalledWith('1', [0])
  })

  it('shows attachment', () => {
    props.reply = Object.assign(props.reply, { attachment: { } })
    let tree = shallow(<Reply {...props}/>)
    let showAttachment = tree.find(`[testID="discussion-reply.${props.reply.id}.attachment"]`)
    showAttachment.simulate('press')

    expect(props.navigator.show).toHaveBeenCalledWith(
      '/attachment',
      { modal: true },
      { attachment: props.reply.attachment }
    )
  })

  describe('ratings', () => {
    beforeEach(() => {
      props.reply = template.discussionReply({ id: '1', rating_sum: 2 })
    })

    it('renders rating', () => {
      props.showRating = true
      props.canRate = true
      let tree = shallow(<Reply {...props} />)
      expect(tree.find('[testID="discussion.reply.rating-count"]').length).toEqual(1)
    })

    it('renders user rating', () => {
      props.showRating = true
      props.canRate = true
      props.rating = 1
      let tree = shallow(<Reply {...props} />)
      let rating = tree.find('[testID="discussion.reply.rating-count"]')
      expect(rating.props().children).toEqual(['(', '2', ')'])
    })

    it('renders rating when user cant rate', () => {
      props.showRating = true
      props.canRate = false
      let tree = shallow(<Reply {...props} />)
      let rating = tree.find('[testID="discussion.reply.rating-count"]')
      expect(rating.props().children).toEqual(['(', '2 likes', ')'])
    })

    it('renders rating after user rates for first time', () => {
      props.showRating = true
      props.canRate = true
      props.rating = null
      const view = shallow(<Reply {...props }/>)
      const rateBtn = view.find('[testID="discussion.reply.rate-btn"]')
      rateBtn.simulate('Press')
      view.update()
      expect(view).toMatchSnapshot()
    })

    it('renders rating after user updates rating', () => {
      props.showRating = true
      props.canRate = true
      props.rating = 1
      const view = shallow(<Reply {...props }/>)
      const rateBtn = view.find('[testID="discussion.reply.rate-btn"]')
      rateBtn.simulate('Press')
      view.update()
      expect(view).toMatchSnapshot()
    })

    it('fixes unverified urls', async () => {
      const evaluateJavaScript = jest.fn()
      const url = 'https://canvas.instructure.com/files/1/preview?verifier=1234'
      const promise = Promise.resolve({ data: template.file({ url }) })
      httpClient().get = jest.fn(() => promise)
      let imageReply = template.discussionReply({ id: '1', message: `<img src="${url}" />` })

      const screen = shallow(<Reply {...props} reply={imageReply} />)
      const webView = screen.find('CanvasWebView')
      webView.getElement().ref({ evaluateJavaScript })

      webView.prop('onFinishedLoading')()
      expect(evaluateJavaScript).toHaveBeenCalled()
      expect(evaluateJavaScript.mock.calls).toMatchSnapshot()

      evaluateJavaScript.mockClear()

      const message = { type: 'BROKEN_IMAGES', data: ['api-url'] }
      webView.prop('onMessage')({ body: JSON.stringify(message) })
      await promise
      expect(evaluateJavaScript).toHaveBeenCalled()
      expect(evaluateJavaScript.mock.calls).toMatchSnapshot()
    })
  })
})
