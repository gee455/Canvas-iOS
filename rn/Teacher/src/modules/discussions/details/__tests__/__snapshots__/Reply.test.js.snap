// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionReplies ratings fixes unverified urls 1`] = `
Array [
  Array [
    ";(() => {
      let imageFiles = Array.from(document.querySelectorAll('img[data-api-returntype=\\"File\\"]'))
      let unverified = imageFiles.filter(img => !(/verifier=/.test(img.src))).map(img => img.dataset.apiEndpoint)
      window.webkit.messageHandlers.canvas.postMessage(JSON.stringify({ type: 'BROKEN_IMAGES', data: unverified }))
    })()",
  ],
]
`;

exports[`DiscussionReplies ratings fixes unverified urls 2`] = `
Array [
  Array [
    ";(() => {
            let image = document.querySelector('img[data-api-endpoint=\\"api-url\\"]')
            if (image) image.src = 'https://canvas.instructure.com/files/1/preview?verifier=1234'
          })()",
  ],
]
`;

exports[`DiscussionReplies ratings renders rating after user rates for first time 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <ThreadedLinesView
    avatarSize={24}
    depth={0}
    marginRight={8}
    reply={
      Object {
        "created_at": "2017-05-23T17:12:04Z",
        "id": "1",
        "message": "<p>1.0</p>",
        "parent_id": null,
        "rating_count": null,
        "rating_sum": 2,
        "replies": Array [],
        "updated_at": "2017-05-23T17:12:04Z",
        "user_id": "1",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginRight": 8,
        "marginTop": 10,
        "paddingTop": 6,
      }
    }
  >
    <View />
    <Avatar
      avatarURL="http://www.fillmurray.com/100/100"
      height={24}
      key="1"
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 12,
          "height": 24,
          "marginTop": 12.8,
          "width": 24,
        }
      }
      testID="reply.avatar"
      userName="The Donald"
    />
    <View
      style={
        Object {
          "backgroundColor": "#f5f5f5",
          "flex": 1,
          "width": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginTop": 12.8,
        }
      }
    >
      <Text
        accessibilityTraits="none"
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        The Donald
      </Text>
      <Text
        style={
          Object {
            "color": "#73818C",
            "fontSize": 12,
            "marginBottom": 16,
          }
        }
      >
        May 23 at 11:12 AM
      </Text>
      <RichContent
        html="<p>1.0</p>"
        navigator={
          Object {
            "dismiss": [Function],
            "dismissAllModals": [Function],
            "isModal": false,
            "pop": [Function],
            "show": [Function],
            "traitCollection": [Function],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <LinkButton
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "paddingBottom": 2,
                "paddingRight": 2,
                "paddingTop": 2,
              }
            }
            testID="discussion.reply-btn"
            textAttributes={
              Object {
                "color": "#8B969E",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Reply
          </LinkButton>
          <Text
            accessible={false}
            style={
              Array [
                Object {
                  "alignSelf": "flex-end",
                  "paddingBottom": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#C7CDD1",
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "textAlign": "center",
                },
              ]
            }
          >
            |
          </Text>
          <LinkButton
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "paddingBottom": 2,
                "paddingRight": 2,
                "paddingTop": 2,
              }
            }
            testID="discussion.edit-btn"
            textAttributes={
              Object {
                "color": "#8B969E",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Edit
          </LinkButton>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8B969E",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "color": "#374A59",
                  "marginRight": 6,
                },
              ]
            }
            testID="discussion.reply.rating-count"
          >
            (
            3
            )
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            testID="discussion.reply.rate-btn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/images/discussions/Liked.png",
                }
              }
              style={
                Array [
                  Object {
                    "tintColor": "#8B969E",
                  },
                  Object {
                    "tintColor": "#374A59",
                  },
                ]
              }
            />
          </TouchableOpacity>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;

exports[`DiscussionReplies ratings renders rating after user updates rating 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <ThreadedLinesView
    avatarSize={24}
    depth={0}
    marginRight={8}
    reply={
      Object {
        "created_at": "2017-05-23T17:12:04Z",
        "id": "1",
        "message": "<p>1.0</p>",
        "parent_id": null,
        "rating_count": null,
        "rating_sum": 2,
        "replies": Array [],
        "updated_at": "2017-05-23T17:12:04Z",
        "user_id": "1",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginRight": 8,
        "marginTop": 10,
        "paddingTop": 6,
      }
    }
  >
    <View />
    <Avatar
      avatarURL="http://www.fillmurray.com/100/100"
      height={24}
      key="1"
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 12,
          "height": 24,
          "marginTop": 12.8,
          "width": 24,
        }
      }
      testID="reply.avatar"
      userName="The Donald"
    />
    <View
      style={
        Object {
          "backgroundColor": "#f5f5f5",
          "flex": 1,
          "width": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginTop": 12.8,
        }
      }
    >
      <Text
        accessibilityTraits="none"
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        The Donald
      </Text>
      <Text
        style={
          Object {
            "color": "#73818C",
            "fontSize": 12,
            "marginBottom": 16,
          }
        }
      >
        May 23 at 11:12 AM
      </Text>
      <RichContent
        html="<p>1.0</p>"
        navigator={
          Object {
            "dismiss": [Function],
            "dismissAllModals": [Function],
            "isModal": false,
            "pop": [Function],
            "show": [Function],
            "traitCollection": [Function],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <LinkButton
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "paddingBottom": 2,
                "paddingRight": 2,
                "paddingTop": 2,
              }
            }
            testID="discussion.reply-btn"
            textAttributes={
              Object {
                "color": "#8B969E",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Reply
          </LinkButton>
          <Text
            accessible={false}
            style={
              Array [
                Object {
                  "alignSelf": "flex-end",
                  "paddingBottom": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#C7CDD1",
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "textAlign": "center",
                },
              ]
            }
          >
            |
          </Text>
          <LinkButton
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "paddingBottom": 2,
                "paddingRight": 2,
                "paddingTop": 2,
              }
            }
            testID="discussion.edit-btn"
            textAttributes={
              Object {
                "color": "#8B969E",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Edit
          </LinkButton>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#8B969E",
                  "fontSize": 16,
                  "fontWeight": "500",
                },
                Object {
                  "color": "#8B969E",
                  "marginRight": 6,
                },
              ]
            }
            testID="discussion.reply.rating-count"
          >
            (
            1
            )
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            testID="discussion.reply.rate-btn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/images/discussions/Like.png",
                }
              }
              style={
                Array [
                  Object {
                    "tintColor": "#8B969E",
                  },
                  Object {
                    "tintColor": "#8B969E",
                  },
                ]
              }
            />
          </TouchableOpacity>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;

exports[`DiscussionReplies renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <ThreadedLinesView
    avatarSize={24}
    depth={0}
    marginRight={8}
    reply={
      Object {
        "created_at": "2017-05-23T17:12:04Z",
        "id": "1",
        "message": "<p>1.0</p>",
        "parent_id": null,
        "rating_count": null,
        "rating_sum": null,
        "replies": Array [
          Object {
            "created_at": "2017-05-23T17:12:04Z",
            "id": 2,
            "message": "<p>1.0</p>",
            "parent_id": null,
            "rating_count": null,
            "rating_sum": null,
            "replies": Array [],
            "updated_at": "2017-05-23T17:12:04Z",
            "user_id": "1",
          },
        ],
        "updated_at": "2017-05-23T17:12:04Z",
        "user_id": "1",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginRight": 8,
        "marginTop": 10,
        "paddingTop": 6,
      }
    }
  >
    <View />
    <Avatar
      avatarURL="http://www.fillmurray.com/100/100"
      height={24}
      key="1"
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 12,
          "height": 24,
          "marginTop": 12.8,
          "width": 24,
        }
      }
      testID="reply.avatar"
      userName="The Donald"
    />
    <View
      style={
        Object {
          "backgroundColor": "#f5f5f5",
          "flex": 1,
          "width": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "marginTop": 12.8,
        }
      }
    >
      <Text
        accessibilityTraits="none"
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        The Donald
      </Text>
      <Text
        style={
          Object {
            "color": "#73818C",
            "fontSize": 12,
            "marginBottom": 16,
          }
        }
      >
        May 23 at 11:12 AM
      </Text>
      <RichContent
        html="<p>1.0</p>"
        navigator={
          Object {
            "dismiss": [Function],
            "dismissAllModals": [Function],
            "isModal": false,
            "pop": [Function],
            "show": [Function],
            "traitCollection": [Function],
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 8,
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <LinkButton
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "paddingBottom": 2,
                "paddingRight": 2,
                "paddingTop": 2,
              }
            }
            testID="discussion.reply-btn"
            textAttributes={
              Object {
                "color": "#8B969E",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Reply
          </LinkButton>
          <Text
            accessible={false}
            style={
              Array [
                Object {
                  "alignSelf": "flex-end",
                  "paddingBottom": 2,
                  "paddingRight": 2,
                  "paddingTop": 2,
                },
                Object {
                  "alignSelf": "center",
                  "color": "#C7CDD1",
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "textAlign": "center",
                },
              ]
            }
          >
            |
          </Text>
          <LinkButton
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "paddingBottom": 2,
                "paddingRight": 2,
                "paddingTop": 2,
              }
            }
            testID="discussion.edit-btn"
            textAttributes={
              Object {
                "color": "#8B969E",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
            Edit
          </LinkButton>
        </View>
      </View>
      <View />
    </View>
  </View>
</View>
`;
